File: MyWork/app/src/main/java/com/leudroid/mywork/MainActivity.java
package com.leudroid.mywork;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {

    ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.list_view);
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
                android.R.layout.simple_list_item_1,
                getResources().getStringArray(R.array.modules));
        listView.setAdapter(adapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                switch (position){
                    case 0 :
                        Intent picassoIntent = new Intent(MainActivity.this,PicassoImageLoading.class);
                        startActivity(picassoIntent);
                        break;
                    case 1 :
                        Intent picassoCacheIntent = new Intent(MainActivity.this,ImageCaching.class);
                        startActivity(picassoCacheIntent);
                        break;
                    case 2 :
                        Intent recyclerIntent = new Intent(MainActivity.this,RecyclerActivity.class);
                        startActivity(recyclerIntent);
                        break;
                }
            }
        });


    }
}

File: Code Review/MistplayChallenge/app/src/main/java/com/leudroid/mistplaychallenge/MainActivity.kt
package com.leudroid.mistplaychallenge

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.leudroid.mistplaychallenge.Adapaters.CategoryAdapter
import com.leudroid.mistplaychallenge.POJO.Category
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //Getting Data from Json String and setting up the View
        getView()
    }

    private fun getView() {

        recyclerView = recyclerViewCategory

        //val jsonString = GetData.getJson();
        //val categoryList = object : TypeToken<List<Category>>() { }.type

        // using Gson parsing json string to category object
        val categoryList = Gson().fromJson<List<Category>>(GetData.getJson(),
            object : TypeToken<List<Category>>() { }.type)


        recyclerView.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = CategoryAdapter(categoryList)
        }
    }
}